src/evm/protocol/uniswap_v4/tycho_decoder.rs

```diff
@@ -0,0 +1,159 @@
+use std::collections::HashMap;
+
+use alloy_primitives::U256;
+use tycho_client::feed::{synchronizer::ComponentWithState, Header};
+use tycho_core::Bytes;
+
+use super::state::UniswapV4State;
+use crate::{
+    evm::protocol::{
+        uniswap_v4::state::UniswapV4Fees,
+        utils::uniswap::{i24_be_bytes_to_i32, tick_list::TickInfo},
+    },
+    models::Token,
+    protocol::{errors::InvalidSnapshotError, models::TryFromWithBlock},
+};
+
+impl TryFromWithBlock<ComponentWithState> for UniswapV4State {
+    type Error = InvalidSnapshotError;
+
+    /// Decodes a `ComponentWithState` into a `UniswapV4State`. Errors with a `InvalidSnapshotError`
+    /// if the snapshot is missing any required attributes.
+    fn try_from_with_block(
+        snapshot: ComponentWithState,
+        _block: Header,
+        _all_tokens: &HashMap<Bytes, Token>,
+    ) -> Result<Self, Self::Error> {
+        let liq = snapshot
+            .state
+            .attributes
+            .get("liquidity")
+            .ok_or_else(|| InvalidSnapshotError::MissingAttribute("liquidity".to_string()))?
+            .clone();
+
+        // This is a hotfix because if the liquidity has never been updated after creation, it's
+        // currently encoded as H256::zero(), therefore, we can't decode this as u128.
+        // We can remove this once it has been fixed on the tycho side.
```

Comment:
How do we fix this tycho-side? Is this a substreams, or indexer change?

Reply:
This is copied over from UniV3 implementation. Since liquidity tracking was not changed there, I kept the warning, this is on my TODO to see how can we fix

Reply:
I think this is old and was fixed in the logs only implementation already

Reply:
FYI, I just checked and the issue came from [here](https://github.com/propeller-heads/tycho-indexer/blob/9b8014544bb31b57a198235bcfc3ae0c48cafae7/tycho-indexer/src/extractor/post_processors/attributes.rs#L24C65-L24C82)

This was fixed on the substreams side, so I think we can remove this and below as well